src/CodeGen/ODataT4CodeGenerator.ttinclude

```diff
@@ -1472,7 +1474,14 @@ public abstract class ODataClientTemplate : TemplateBase
             }
         }
 
-        this.WriteGeneratedEdmModel(Utils.SerializeToString(this.context.Edmx).Replace("\"", "\"\""));
+		if(useTempFile) 
+		{
+		  this.WriteGeneratedEdmModel(Utils.SerializeToString(this.context.Edmx));
+		} 
+		else 
+		{
+		  this.WriteGeneratedEdmModel(Utils.SerializeToString(this.context.Edmx).Replace("\"", "\"\""));
+		}

```

Comment:
Instead of doing this useTempFile check here, can you do this inside WriteGeneratedModel? Pass the this.context.Edmx string directly and then decide.

Reply:
String parsed in the WriteGeneratedModel function where the decision now takes place.